django/test/client.py

```diff
@@ -544,7 +544,9 @@ def generic(
                 (b'content-type', content_type.encode('ascii')),
             ])
             s['_body_file'] = FakePayload(data)
-        s.update(extra)
+        s['headers'] += [(key.lower().encode(), value.encode()) for key, value in extra.items() if key != 'follow']
+        if 'follow' in extra:
+            s['follow'] = extra['follow']
```

Comment:
The `follow` kwarg is rejected in `AsyncClient.request()`, which is called by `generic` here. 

Happy to take suggestions.

Reply:
```python
follow = extra.pop('follow', None)
if follow is not None:
    s['follow'] = follow
s['headers'].extend((key.lower().encode(), value.encode()) for key, value in extra.items())
```

Reply:
@kwist-sgr That's better yes. I updated. Thanks. 

It's still a little iffy having to deal with `follow` in two places, but adjusting that is probably out of scope for here… 

👍

Reply:
Or maybe it would be better to extract `follow` from `extra` like this:

```python
def generic(
        self, method, path, data='', content_type='application/octet-stream',
        secure=False, follow=False, **extra,
    ):
    ...
    if follow:
        s['follow'] = follow
    s['headers'] += [(key.lower().encode(), value.encode()) for key, value in extra.items() if key]
```